shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm

```diff
@@ -757,16 +763,11 @@ static bool ClipRRectContainsPlatformViewBoundingRect(const SkRRect& clip_rrect,
   // Mark all layers as available, so they can be used in the next frame.
   std::vector<std::shared_ptr<FlutterPlatformViewLayer>> unused_layers =
       layer_pool_->GetUnusedLayers();
-  FML_DCHECK(unused_layers.empty() || missing_layer_count == 0);
   layer_pool_->RecycleLayers();
 
-  auto task = fml::MakeCopyable([&, platform_view_layers = std::move(platform_view_layers),
-                                 missing_layer_count,                                       //
-                                 current_composition_params = current_composition_params_,  //
-                                 views_to_recomposite = views_to_recomposite_,              //
-                                 callbacks = callbacks,                                     //
-                                 composition_order = composition_order_,                    //
-                                 unused_layers = unused_layers]() mutable {
+  auto task = [this, platform_view_layers = std::move(platform_view_layers),
+               callbacks = std::move(callbacks),
+               unused_layers = std::move(unused_layers)]() mutable {
     TRACE_EVENT0("flutter", "FlutterPlatformViewsController::SubmitFrame::CATransaction");
```

Comment:
We're capturing this already and calling methods on it (CompositeWithParams, RemoveUnusedLayers) which modifies variables such as `views_to_recomposite_` - holding on to previous value in capture causes crashes when removing views.

We may want to synchronize this (i.e. block raster thread until the block completes) - in case there is something holding the platform for longer than expected we can have race condition.